core/src/main/java/hudson/ProxyConfiguration.java

```diff
@@ -146,7 +147,7 @@ private Authenticator newAuthenticator() {
             public PasswordAuthentication getPasswordAuthentication() {
                 String userName = getUserName();
                 if (getRequestorType() == RequestorType.PROXY && userName != null) {
-                    return new PasswordAuthentication(userName, getPassword().toCharArray());
+                    return new PasswordAuthentication(userName, Secret.toString(secretPassword).toCharArray());
```

Comment:
`Secret.toString` may return null in this case after your constructor change, and it will lead to NPE

Reply:
It returns an empty string not null